morph_tool/resampling.py

```diff
@@ -0,0 +1,201 @@
+
+    where (v'[k], v[k]) can be either `points` or any other property mapped
+    on `points`.
+
+    Args:
+        points (np.ndarray): (N, 3) Array of consecutive points definining
+        segments.
+        linear_density (float): Target number of points per micron
+
+    Returns:
+        tuple of numpy arrays:
+            - ids (np.array): (K-2,) Array of positional ids
+            - fractions (np.array): (K-2,) Parametric fractions
+
+    Notes:
+        Given consecutive points [p0, p1, ..., pN] a resampling will create
+        new points on the polyline defined by these points. This means
+        that a new point p' can be defined as a fraction t of an existing
+        segment (p[n], p[n + 1]). Therefore, using the line segment's
+        parametric eq:
+
+        p' = p[n] + t * (p[n+1] - p[n])
+
+        where p[n] is the start of the segment, (p[n+1] - p[n]) the segment's
+        vector and t the fraction [0., 1.] of the vector to create the point.
+
+        Therefore, this function returns K ids and fractions, where each id n
+        and fraction t corespond to tehe equation for p' above. In vectorized
+        form will then be:
+
+        p'[k] = points[ids[k]] + fractions[k] * (p[ids[k] + 1] - p[ids[k]])
+
+        For example if we have four points and the new point p'[k] lies
+        between p[1] and p[2], and exactly at the middle:
+
+        p[0] ---- p[1] --p'[k]-- p[2] ---- p[3]
+
+        then the equation will become:
+
+        p'[k] = p[1] + 0.5 * (p[2] - p[1])
+
+        This interpolation relation holds for all values mapped onto points.
+        For example the respective new diameters d' for i will be:
+
+        d'[k] = d[ids[k]] + fractions[k] * (d[ids[k] + 1] - d[ids[k]])
+    """
+    path_lengths = _vertex_path_lengths(points)
+    total_length = path_lengths[-1]
+
+    # segment number according to linear density, minimum 1
+    n_segments = max(1, int(total_length * linear_density))
+
+    # split total length in n_segments equal parts of dl length
+    dl = total_length / n_segments
+
+    # vertex path lengths without including first and last
+    new_path_lengths = np.arange(dl, total_length, dl)
+
+    # ids of the starting points for the parametric equation
+    ids = np.searchsorted(path_lengths, new_path_lengths) - 1
+
+    # each fraction t corresponds to the fraction of the vector
+    # in the parametric equation of the line segment p0 + t * (p1 - p0)
+    fractions = (new_path_lengths - path_lengths[ids]) / (path_lengths[ids + 1] - path_lengths[ids])
+
+    return ids, fractions
+
+
+def _parametric_values(values, ids, fractions):
+    """It creates a new array of values, where the start and end
+    values remain unchanged and the in-between values are calculated
+    from the parametric equation:
+
+    v'[k] = v[ids[k]] + fractions[k] * (v[ids[k] + 1] - v[ids[k]])
+
+    Args:
+        values (np.ndarray): 1D or 2D array of values
+        ids (np.ndarray): (N,) Array of ints
```

Comment:
"values" taken apart, these could be more detailed (you may also refer to the doc of another function).